src/ConnectedMode.UnitTests/Migration/MigrationSettingsProviderTests.cs

```diff
@@ -41,13 +47,47 @@ public void MefCtor_CheckTypeIsNonShared()
         [TestMethod]
         public async Task Get_ReturnsExpectedValue()
         {
-            var testSubject = CreateTestSubject();
+            var solution = CreateIVsSolution("c:\\rootfolder");
+            var serviceProvider = CreateServiceProviderWithSolution(solution.Object);
 
-            // TODO - fix once implemented
-            var actual = await testSubject.GetAsync("expected project key");
-            actual.Should().BeNull();
+            var testSubject = CreateTestSubject(serviceProvider.Object);
+
+            var actual = await testSubject.GetAsync("slvs_samples_bound_vs2019");
+            actual.Should().NotBeNull();
+            actual.LegacySonarLintFolderPath.Should().Be("c:\\rootfolder\\.sonarlint");
+            actual.PartialCSharpRuleSetPath.Should().Be(".sonarlint\\slvs_samples_bound_vs2019csharp.ruleset");
+            actual.PartialCSharpSonarLintXmlPath.Should().Be(".sonarlint\\slvs_samples_bound_vs2019\\CSharp\\SonarLint.xml");
+            actual.PartialVBRuleSetPath.Should().Be(".sonarlint\\slvs_samples_bound_vs2019vb.ruleset");
+            actual.PartialVBSonarLintXmlPath.Should().Be(".sonarlint\\slvs_samples_bound_vs2019\\VB\\SonarLint.xml");
         }
 
-        private static MigrationSettingsProvider CreateTestSubject() => new MigrationSettingsProvider();
+        private static MigrationSettingsProvider CreateTestSubject(IServiceProvider serviceProvider = null,
+            IThreadHandling threadHandling = null)
+        {
+            serviceProvider ??= Mock.Of<IServiceProvider>();
+            threadHandling ??= new NoOpThreadHandler();
+
+            return new MigrationSettingsProvider(serviceProvider, threadHandling);
+        }
+
+        private static Mock<IServiceProvider> CreateServiceProviderWithSolution(IVsSolution solution)
```

Comment:
This test helper and the one in line 81 are copied from the `FileProviderTests`. I`m not sure it's worth the effort of deduplicating them, though.